internal/rukpak/source/unpacker.go

```diff
@@ -2,10 +2,28 @@ package source
 
 import (
 	"context"
-	"fmt"
 	"io/fs"
 )
 
+// SourceTypeImage is the identifier for image-type bundle sources
+const SourceTypeImage SourceType = "image"
+
+type ImageSource struct {
+	// Ref contains the reference to a container image containing Bundle contents.
+	Ref string
+}
+
+// Unrecoverable represents an error that can not be recovered
+// from without user intervention. When this error is returned
+// the request should not be requeued.
+type Unrecoverable struct {
+	error
+}
```

Comment:
Is there an explicit check for this type of error?

Reply:
Hmm. It appears that we do not check for this error type in our reconciler. I brought this over from the now-deleted `image_registry.go` where it was used.

I do check for it in unit tests.

WDYT about addressing this in a follow-up?

Reply:
Depends on how many other changes will occur if you update... but at least it's isolated.